connectors/confluence/connector/src/main/java/org/apache/manifoldcf/crawler/connectors/confluence/client/ConfluenceClient.java

```diff
@@ -0,0 +1,717 @@
+			HttpResponse response = executeRequest(httpGet);
+			HttpEntity entity = response.getEntity();
+			MutableAttachment attachment = attachmentFromHttpEntity(entity);
+			EntityUtils.consume(entity);
+			retrieveAndSetAttachmentContent(attachment);
+			return attachment;
+		} catch (Exception e) {
+			logger.error("[Processing] Failed to get attachment {}. Error: {}",
+					url, e.getMessage());
+		}
+
+		return new Attachment();
+	}
+
+	/**
+	 * <p>
+	 * Downloads and retrieves the attachment content, setting it in the given
+	 * {@code Attachment} instance
+	 * </p>
+	 * 
+	 * @param attachment
+	 *            the {@code Attachment} instance to download and set the
+	 *            content
+	 * @throws Exception
+	 */
+	private void retrieveAndSetAttachmentContent(MutableAttachment attachment)
+			throws Exception {
+		StringBuilder sb = new StringBuilder();
+		sb.append(attachment.getBaseUrl()).append(attachment.getUrlContext())
+				.append(attachment.getDownloadUrl());
+		String url = sanitizeUrl(sb.toString());
+		logger.debug(
+				"[Processing] Hitting url for getting attachment content : {}",
+				url);
+		try {
+			HttpGet httpGet = createGetRequest(url);
+			HttpResponse response = executeRequest(httpGet);
+			attachment.setLength(response.getEntity().getContentLength());
+			byte[] byteContent = IOUtils.toByteArray(response.getEntity()
+					.getContent());
+			EntityUtils.consumeQuietly(response.getEntity());
+			attachment.setContentStream(new ByteArrayInputStream(byteContent));
+		} catch (Exception e) {
+
+			logger.error(
+					"[Processing] Failed to get attachment content from {}. Error: {}",
+					url, e.getMessage());
+			throw e;
+		}
+
+	}
+
+
+	/**
+	 * <p>Get a Confluence page identified by its id</p>
+	 * @param pageId the page id
+	 * @return the Confluence page
+	 */
+	public Page getPage(String pageId) {
+		String url = String
+				.format("%s://%s:%s/%s/%s/%s?%s",
+						protocol, host, port, path, CONTENT_PATH, pageId, EXPANDABLE_PARAMETERS);
+		url = sanitizeUrl(url);
+		logger.debug(
+				"[Processing] Hitting url for getting document content : {}",
+				url);
+		try {
+			HttpGet httpGet = createGetRequest(url);
+			HttpResponse response = executeRequest(httpGet);
+			HttpEntity entity = response.getEntity();
+			MutablePage page = pageFromHttpEntity(entity);
+			EntityUtils.consume(entity);
+			List<Label> labels = getLabels(pageId);
+			page.setLabels(labels);
+			return page;
+		} catch (Exception e) {
+			logger.error("[Processing] Failed to get page {0}. Error: {1}",
+					url, e.getMessage());
+		}
+
+		return new Page();
+	}
+
+	/**
+	 * <p>Get the labels of a specific page</p> 
+	 * @param pageId The pageId to get the labels
+	 * @return a {@code List<Label>} of labels
+	 */
+	public List<Label> getLabels(String pageId) {
+				
+		List<Label> labels = Lists.newArrayList();
+		int lastStart = 0;
+		int limit = 50;
+		boolean isLast = false;
+		do {
+			String url = String
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code uses &#39;%s&#39; to format integers: lastStart (declared at line 476), limit. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 481. Consider formatting this integer with &#39;%d&#39; instead.